### ISSUE TITLE
PR GitHub action build: two builds intended?

### ISSUE THREAD

**Issue by plata** [blue-build/template#28]
In a pull request, the GitHub action runs twice (for push and for pull request):
![checks](https://github.com/blue-build/template/assets/3973260/9a9a7dbd-8170-4ceb-aa78-b486df70bf86)
Is this intended?

**Comment by xynydev** [comment]
If you are doing a PR from your own repository, I guess, since it is intended that both internal branches without PRs and external PRs without internal branches run a build. I guess one of those could be disabled in the action, though.

**Comment by plata** [comment]
If you agree to have the push build only on branch master, I can open a PR for it.

**Comment by xynydev** [comment]
No, it is intended that pushes happen on all branches and on PRs. It can be considered a bug or and edge case than on an in-repo PR both builds are triggered, and a PR mitigating that would be merged.

**Comment by plata** [comment]
This doesn't seem to be easy to achieve with the current [workflow syntax](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions). [This proposal](https://github.com/actions/runner/issues/859) would help.

The only possibility apparently is a conditional implementation per job: https://github.com/orgs/community/discussions/57827.

However, if the build runs for every push anyway, why would you even want to have the `pull_request` trigger as well? Is there any possibility to create a pull request change without a push?

**Comment by xynydev** [comment]
> However, if the build runs for every push anyway, why would you even want to have the pull_request trigger as well? Is there any possibility to create a pull request change without a push?

Yes, it is. `push` only runs if the push is into the repository, while `pull_request` also triggers on PRs from forks.

> The only possibility apparently is a conditional implementation per job: https://github.com/orgs/community/discussions/57827.

That is not only the only possibility, but the best one. The action in this repository only has a single job with a single step, so no repetition would be needed.

**Comment by Kabuki94** [comment]
push all pending updates
